crates/polars-io/src/parquet/read/options.rs

```diff
@@ -1,11 +1,12 @@
 use arrow::datatypes::ArrowSchemaRef;
+use polars_core::schema::SchemaRef;
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
 #[derive(Clone, Debug, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub struct ParquetOptions {
-    pub schema: Option<ArrowSchemaRef>,
+    pub schema: Option<SchemaRef>,
```

Comment:
Change `ParquetOptions::schema` to hold our native schema type, this reduces excess copy/conversions as that is the type we receive at input, and also the type we give to the IR. The conversion to arrow  schema is now done below in `to_alp_impl` and stored directly to the `FileInfo::reader_schema` field